be/src/storage/tablet.cpp

```diff
@@ -276,6 +276,7 @@ void Tablet::modify_rowsets(const std::vector<RowsetSharedPtr>& to_add, const st
 
     // add rs_metas_to_delete to tracker
     _timestamped_version_tracker.add_stale_path_version(rs_metas_to_delete);
+    Rowset::close_rowsets(to_delete);
```

Comment:
Actually I don't think this is a good place to close_rowsets.
I think caller should do this things. 
This function just do meta modification. It don't know if the rowsets will be closed.
I think the compaction manager will decide if these rowsets should be closed not this function.